pkg/odo/cli/component/delete.go

```diff
@@ -210,6 +181,44 @@ func (do *DeleteOptions) Run() (err error) {
 	return
 }
 
+// Run has the logic to perform the required actions as part of command for devfiles
+func (do *DeleteOptions) DevFileRun() (err error) {
```

Comment:
@mik-dass Any plan for writing UTs ?

Reply:
It's the CLI package. We are testing CLI by using integration tests I guess.

Reply:
Really i am in 50-50 state whether to write UTs for cli packages or not. Recently i saw UTs for few devfile cli packages, so thought why not then covering UTs for regular cli packages. More importantly our UTs coverage are being run on all regular cli packages having no test file.

Ping @girishramnani  @kadel

Reply:
> Recently i saw UTs for few devfile cli packages, so thought why not then covering UTs for regular cli packages.

TBH I couldn't any. Please share some links.

Reply:
This is a generic topic for cabal discussion i guess. I will create a separate issue to tack the discussion of whether to add UTs for cli packages are sensible or not.   

So atm i am applying lgtm label for your current changes
/lgtm

Reply:
Cool thanks for creating the issue :+1:

Reply:
> TBH I couldn't any. Please share some links. Also is the effort even worth it?

Please refer #2867, there i have added the reference.